ruma-identity-service-api/src/keys/check_public_key_validity/v2.rs

```diff
@@ -0,0 +1,39 @@
+//! [GET /_matrix/identity/v2/pubkey/isvalid](https://matrix.org/docs/spec/identity_service/0.3.0#get-matrix-identity-v2-pubkey-isvalid)
+
+use ruma_api::ruma_api;
+
+ruma_api! {
+    metadata: {
+        description: "Check whether a long-term public key is valid. The response should always be the same, provided the key exists.",
+        method: GET,
+        name: "check_public_key_validity",
+        path: "/_matrix/identity/v2/pubkey/isvalid",
+        authentication: None,
+        rate_limited: false,
+    }
+
+    request: {
+        /// Base64-encoded (no padding) public key to check for validity
+        #[ruma_api(query)]
+        pub public_key: &'a str,
+    }
+
+    response: {
+        /// Whether the public key is recognised and is currently valid.
+        pub valid: bool,
+    }
+}
+
+impl<'a> Request<'a> {
+    /// Create a `Request` with the given paramters.
```

Comment:
Type: `paramters`. Being more specific would be nice though (i.e. "given public key"). Same for the response.